pkg/pointerdb/pointerdb.go

```diff
@@ -53,11 +58,35 @@ func (s *Server) validateAuth(APIKey []byte) error {
 	return nil
 }
 
+func (s *Server) validateSegment(req *pb.PutRequest) error {
+	min := s.config.MinInlineSegmentSize
+	max := s.config.MaxInlineSegmentSize
+	inlineSize := len(req.GetPointer().InlineSegment)
+	remote := req.GetPointer().Remote
+
+	if inlineSize < min {
```

Comment:
wait, i think this should be `if remote != nil && remote.Size < min`. sorry for missing it the first time around